src/Dialect/Krnl/DialectBuilder.hpp

```diff
@@ -95,6 +100,9 @@ struct KrnlBuilder : public DialectBuilder {
   void iterateIE(mlir::ValueRange originalLoops,
       mlir::ValueRange optimizedLoops, mlir::ArrayRef<IndexExpr> lbs,
       mlir::ArrayRef<IndexExpr> ubs, KrnlLoopBodyFn bodyBuilderFn) const;
+  void iterateIE(mlir::ValueRange originalLoops,
```

Comment:
Because we are well acquainted with the normal function, but not so much with the new interface, I think this deserves an explanation here of how adding the second set of ValueRange to the lambda function changes things. Are there cases where both are needed? Or is it solely a matter of distinguishing the old and the new way?

Reply:
The essential idea in krnl is to separate computation and schedule. I feel that the ultimate goal is to express optimizations by schedule ops like block, permute, unroll while keeping the iterate's body unchanged. However, I am not sure at this moment if SIMD can be totally expressed by schedule ops or not. So, during this transition period, I just keep both sets of ValueRange. Users can use one of them or both as long as the generated IR is valid (though I don't have an example of using both). Also, since we have two sets of ValueRange for originalLoop and optimizedLoop, it makes sense to have two sets of ValueRange for their indices, respectively.

Reply:
@AlexandreEichenberger I changed the function name and use a single ValueRange for the original loop, which avoids confusion.